internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go

```diff
@@ -2141,6 +2144,51 @@ func KexecPrepare(_ runtime.Sequence, data any) (runtime.TaskExecutionFunc, stri
 
 		defer kernel.Close() //nolint:errcheck
 
+		fd := int(kernel.Fd())
+
+		if goruntime.GOARCH == "arm64" {
+			// https://git.kernel.org/pub/scm/utils/kernel/kexec/kexec-tools.git/tree/include/kexec-pe-zboot.h
+			var peZbootheaderData [24]byte
+
+			if _, err := kernel.Read(peZbootheaderData[:]); err != nil {
+				return fmt.Errorf("failed to read PE Zboot header: %w", err)
+			}
+
+			if !bytes.Equal(peZbootheaderData[:4], []byte("MZ\x00\x00")) {
+				return fmt.Errorf("invalid PE Zboot header")
+			}
+
+			if !bytes.Equal(peZbootheaderData[4:8], []byte("zimg")) {
```

Comment:
i guess we should skip this if it's not a `zimg`? for upgrades from 1.7

Reply:
that would be a downgrade from 1.8 to 1.7 I guess, as the host Talos is 1.8, but the kernel is not zboot. I guess in this case we could say just skip to plain `kexec`